test/box-luatest/gh_7448_net_box_cancelled_fiber_error_test.lua

```diff
@@ -0,0 +1,31 @@
+local server = require('test.luatest_helpers.server')
+local net = require('net.box')
+local t = require('luatest')
+
+local g = t.group()
+
+g.before_all = function()
+    g.server = server:new{alias = 'default'}
+    g.server:start()
+    g.conn = net.connect(g.server.net_box_uri)
+end
+
+g.after_all = function()
+    g.conn:close()
+    g.server:drop()
+end
+
+g.test_conn_cancelled_fiber_error = function()
+    t.assert(g.conn:ping())
+    -- Explicit kill of the connection's fiber
+    pcall(function()
+        g.conn:eval([[
+            box.session.push(nil)
+        ]], {}, {on_push = function()
+            require('fiber').self():cancel()
+        end})
```

Comment:
It makes sense to explain in a comment that you are using pushes to reach the worker fiber somehow. But otherwise it is not really needed. This test runs only on versions where you have `conn._fiber` available and can do `conn._fiber:cancel()`. In vshard I proposed this push-hack as a way to repeat the same on 1.10 where you don't have `._fiber` member.

Reply:
I'm already used to the fact that we test changes on all versions like in vshard. You're right, there's no need for such workaround here. Fixed